atom/browser/common_web_contents_delegate.cc

```diff
@@ -86,6 +87,35 @@ base::DictionaryValue* CreateFileSystemValue(const FileSystem& file_system) {
   return file_system_value;
 }
 
+void WriteToFile(brightray::InspectableWebContents* web_contents,
+                 const base::FilePath& path,
+                 const std::string& content,
+                 const std::string& url) {
+  DCHECK_CURRENTLY_ON(content::BrowserThread::FILE);
+  DCHECK(!path.empty());
+
+  if (base::WriteFile(path, content.data(), content.size()) ==
+      static_cast<int>(content.size())) {
+    base::StringValue url_value(url);
+    web_contents->CallClientFunction(
+        "DevToolsAPI.savedURL", &url_value, nullptr, nullptr);
+  }
+}
+
+void AppendToFile(brightray::InspectableWebContents* web_contents,
+                  const base::FilePath& path,
+                  const std::string& content,
+                  const std::string& url) {
+  DCHECK_CURRENTLY_ON(content::BrowserThread::FILE);
+  DCHECK(!path.empty());
+
+  if (base::AppendToFile(path, content.data(), content.size())) {
+    base::StringValue url_value(url);
+    web_contents->CallClientFunction(
```

Comment:
`PostTaskAndReply` should be enough, `DevToolsAPI.savedURL` doesn't care whether file is written successfully.